net/discovery.go

```diff
@@ -180,13 +143,17 @@ func (d *discovery) discoverLoop() {
 	}
 }
 
-func (d *discovery) findPeers(provides string, timeout time.Duration) ([]peer.AddrInfo, error) {
-	peerCh, err := d.rd.FindPeers(d.ctx, provides)
+func (d *discovery) findPeers(provides string, timeout time.Duration) ([]peer.ID, error) {
+	peerCh, err := d.rd.FindPeers(d.ctx, provides,
+		libp2pdiscovery.Limit(defaultMaxPeers),
+		libp2pdiscovery.TTL(defaultAdvertiseTTL),
```

Comment:
what does this do?

Reply:
I meant to explicitly ask for your feedback on this one.  I added this code when trying to debug a different problem, but this change wasn't what solved the problem. It should work without it, but it may add value.  Here's the documentation on them:
https://github.com/libp2p/go-libp2p/blob/v0.24.0/core/discovery/options.go#L27-L41

Reply:
I'm wondering if this option actually does anything, looking at what `Advertise` does here it doesn't look like it's actually passed to the DHT: https://github.com/libp2p/go-libp2p/blob/v0.24.1/p2p/discovery/routing/routing.go#L25

Reply:
yes, I think that's good

Reply:
done